core/src/main/java/com/turn/ttorrent/common/Torrent.java

```diff
@@ -676,21 +676,25 @@ private static Torrent create(File parent, List<File> files, int pieceLength,
 
 		private final MessageDigest md;
 		private final ByteBuffer data;
+		private final ByteBufferPool bbp;
 
-		CallableChunkHasher(ByteBuffer buffer) throws NoSuchAlgorithmException {
+		CallableChunkHasher(ByteBuffer rentedBuffer, ByteBufferPool bbp) throws NoSuchAlgorithmException {
```

Comment:
Indeed, the ByteBufferPool is a good solution for the problem described at #35 as well, not thrashing the memory with short-lived objects, which makes it a good thing to not put it inside the hashing loop, but unless I return a tuple as the Callable's type ` class CallableChunkHasher implements Callable<Pair<String, ByteBuffer>` I don't see a way of fixing this assymetry, quite frankly I don't know what's better, of if there's a better way that I haven't seen